src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/ArrayBufferWriter.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json.Serialization
+{
+    // Note: this is an internal class that will be replaced with a shared version.
+    internal class ArrayBufferWriter : IBufferWriter<byte>, IDisposable
```

Comment:
Maybe use the implementation we are actually shipping and get rid of ResizableArray: https://github.com/dotnet/corefx/blob/4e715d0ba37c3775e3f10ac4741264f556b84684/src/System.Buffers/src/System/Buffers/ArrayBufferWriter.cs

Reply:
OK thanks I'll use the current code.